packages/common/src/utils/domUtils.ts

```diff
@@ -134,38 +140,43 @@ export function setControlValue(
   el: FormControl,
   value: FieldValue | FieldValue[]
 ): void {
-  if (!isInputElement(el)) return;
+  if (!isFormControl(el)) return;
   const fieldValue = value;
-  if (el.type === 'checkbox') {
-    const checkboxesDefaultData = fieldValue;
-    if (
-      typeof checkboxesDefaultData === 'undefined' ||
-      typeof checkboxesDefaultData === 'boolean'
-    ) {
-      el.checked = !!checkboxesDefaultData;
+
+  if (isInputElement(el)) {
+    if (el.type === 'checkbox') {
+      const checkboxesDefaultData = fieldValue;
+      if (
+        typeof checkboxesDefaultData === 'undefined' ||
+        typeof checkboxesDefaultData === 'boolean'
+      ) {
+        el.checked = !!checkboxesDefaultData;
+        return;
+      }
+      if (Array.isArray(checkboxesDefaultData)) {
+        if ((checkboxesDefaultData as string[]).includes(el.value)) {
+          el.checked = true;
+        } else {
+          el.checked = false;
+        }
+      }
       return;
     }
-    if (Array.isArray(checkboxesDefaultData)) {
-      if ((checkboxesDefaultData as string[]).includes(el.value)) {
-        el.checked = true;
-      } else {
-        el.checked = false;
-      }
+    if (el.type === 'radio') {
+      const radioValue = fieldValue;
+      if (el.value === radioValue) el.checked = true;
+      else el.checked = false;
+      return;
     }
-    return;
-  }
-  if (el.type === 'radio') {
-    const radioValue = fieldValue;
-    if (el.value === radioValue) el.checked = true;
-    else el.checked = false;
-    return;
-  }
-  if (el.type === 'file') {
-    el.files = null;
-    el.value = '';
-    return;
+    if (el.type === 'file') {
+      el.files = null;
+      el.value = '';
+      return;
+    }
+    el.value = String(fieldValue || '');
+  } else if (isSelectElement(el) || isTextAreaElement(el)) {
+    el.value = String(fieldValue || '');
```

Comment:
I feel this lines can remain as the original line 168, since the behaviour is the same for all the rest of cases (text input, select input and text area).

Reply:
I get these if I remove the conditions.
![image](https://user-images.githubusercontent.com/5698706/121915169-983acc00-cd50-11eb-8464-7c5bfa9db6c5.png)

Which means there has to be type guards somewhere.

Reply:
I moved the `<input>` specific part within the type guard. The rest doesn't have a check.